### DISCUSSION TITLE
Which version to use if I am on NextJS v13.1.0?

### DISCUSSION THREAD

**Discussion by alan-automationworks** [47ng/nuqs#371] [General] [answered]
I cannot yet upgrade to NextJS version 13.4, and I'd like to use this library. Which version should I use if I am on 13.1.0 using the `app` folder?

**Comment by franky47** [comment]
I'm not sure if 13.1 has the `next/navigation` router API with `useSearchParams`, that's what we use.

If you go for the latest version and ignore peerDependencies warnings, does it work?

**Reply by alan-automationworks** [reply]
It has support for `useSearchParams`. I'll give it a shot and let you know.

**Reply by franky47** [reply, marked as answer]
I ran the e2e test suite against Next.js 13.1.0 and 13.1.5 using the latest next-usequerystate (1.12.0), it works fine (both in app and pages routers), apart from the scroll option not being available on the app router interface in those versions of Next.js, so scroll control may not work as described in the docs.
